src/test/java/com/puppycrawl/tools/checkstyle/bdd/InputConfiguration.java

```diff
@@ -0,0 +1,112 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2021 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.bdd;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+
+public final class InputConfiguration {
+
+    /** The check name. */
+    private final String checkName;
+
+    /** Map of default properties. */
+    private final Map<String, String> defaultProperties;
+
+    /** Map of non default properties. */
+    private final Map<String, String> nonDefaultProperties;
+
+    /** List of violations. */
+    private final List<Integer> violations;
+
+    private InputConfiguration(String checkName,
+                               Map<String, String> defaultProperties,
+                               Map<String, String> nonDefaultProperties,
+                               List<Integer> violations) {
+        this.checkName = checkName;
+        this.defaultProperties = defaultProperties;
+        this.nonDefaultProperties = nonDefaultProperties;
+        this.violations = violations;
+    }
+
+    public String getCheckName() {
+        return checkName;
+    }
+
+    public Map<String, String> getAllProperties() {
+        final Map<String, String> properties = new HashMap<>();
+        properties.putAll(defaultProperties);
+        properties.putAll(nonDefaultProperties);
+        return Collections.unmodifiableMap(properties);
+    }
+
+    public Map<String, String> getDefaultProperties() {
+        return Collections.unmodifiableMap(defaultProperties);
+    }
+
+    public Map<String, String> getNonDefaultProperties() {
+        return Collections.unmodifiableMap(nonDefaultProperties);
+    }
+
+    public List<Integer> getViolations() {
```

Comment:
Configuration is for Check creation, Violations are from runtime of Check instance.

it is not logical to combine all in one class. Violations should be moved out of this class.

Reply:
I think the idea @timurt suggested was to have a data structure like class to store all data from an input file. In that case this should work fine. Don't you think so?

Reply:
@romani 
> it is not logical to combine all in one class. Violations should be moved out of this class.

We can iterate input file twice and have separate method for violation list extraction, but it seems redundant overhead

Also this class is not `DefaultConfiguration`, it is temporal data structure, so I do not see problems with storing violations inside

Reply:
Ok let's keep it for now.